rfc/system/1855-type-metadata.md

```diff
@@ -0,0 +1,130 @@
+# [RR] Typing Metadata
+
+**Authors:**
+
+- @maximsmol
+- @kennyworkman
+
+## 1 Executive Summary
+
+A small patch to introduce flytekit support for python-native type metadata.
+
+## 2 Motivation
+
+Users of flyte wish to have access to arbitrary parameter metadata specified in
+typing annotations.
+
+The `LiteralType` definition in flyteidl already has a `metadata` field in its
+message definition
+(https://github.com/flyteorg/flyteidl/blob/master/protos/flyteidl/core/types.proto#L89),
+suggesting that support for metadata annotations for flyte types was in the
+works.
+
+We wish to add python-native support for this field by allowing flytekit to
+recognize and parse the use of
+[`typing.Annotation`](https://docs.python.org/3/library/typing.html#typing.Annotated).
+
+This could be to change the presentation of a form input ingesting typed values,
+surface descriptions to users or store indexes to control the order of
+parameters with custom frontends.
+
+This feature will not send metadata to the container at runtime when a task is
+invoked.  If the container has access to the original code, it will be able to
+access/load metadata from that. This is particularly important to note for
+TaskTemplate-based flytekit plugins.
+
+An example of how we would use this annotation to tag a parameter with
+presentation information:
+
+
+```
+amplicon_min_alignment_score: Optional[
+        List[
+            Annotated[
+                int,
+                FlyteMetadata(
+                    {
+                        "appearance": {
+                            "multiselect": {
```

Comment:
i this example, should this not be an enum instead?

Reply:
or is this purely cosmetic?

Reply:
Also if we are storing this information, its more than just metadata - atleast in this example, it seems to be validation fields?

Reply:
this is purely cosmetic. it will be used by a custom frontend to saturate component props in this example